pkg/lib/operation/template/local/test/operation.go

```diff
@@ -321,7 +324,40 @@ func runRemoteTest(testName string, tmpl *template.Template, repoFS filesystem.F
 		LogUntrackedPaths: true,
 		ChangeDescription: "",
 	}
-	return syncPush.Run(prjState, pushOpts, projectDeps)
+	err = syncPush.Run(prjState, pushOpts, projectDeps)
+	if err != nil {
+		return err
+	}
+
+	// Get mainConfig from template
+	tickets := storageapi.NewTicketProvider(d.Ctx(), testPrj.StorageAPIClient())
+	ctx := use.NewContext(d.Ctx(), tmpl.Reference(), tmpl.ObjectsRoot(), tmplInstID, branchKey, inputValues, tmpl.Inputs().InputsMap(), tickets)
+	tmplState, err := tmpl.LoadState(ctx, loadState.Options{
+		LoadLocalState:    true,
+		LoadRemoteState:   false,
+		IgnoreNotFoundErr: false,
+	})
```

Comment:
@JakubMatejka myslim, ze problem bude tu ... pri nacitani sablony sa vzdy vygeneruju nove IDs ... nemozes to nacitavat zo sablony, pozri si ako sa main config nacitava v API:
https://github.com/keboola/keboola-as-code/blob/48005a58cb92f96c873f269dc5f5bd8dea223240/internal/pkg/api/server/templates/service/mapper.go#L263-L271

Reply:
🤔 ahá, takže to mám načíst z metadat instance po aplikování šablony! Ok, díky